commit 53e9b1de6840d9047f768878adcbd1d116f72aca
Author: David Gnedt <david.gnedt@davizone.at>
Date:   Mon Jul 19 20:44:02 2010 +0200

    mac80211: set carrier on for monitor interfaces on ieee80211_open
    
    If a station interface is reused as monitor interface it is possible that
    the carrier is still set to off. This breaks packet injection on that
    monitor interface.
    Force the carrier on in monitor interface initialisation like it is also done
    for other interface types (e.g. adhoc, mesh point, ap).
    
    Signed-off-by: David Gnedt <david.gnedt@davizone.at>
    Acked-by: Johannes Berg <johannes@sipsolutions.net>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/net/mac80211/iface.c b/net/mac80211/iface.c
index 8ef2fde6e920..ebbe264e2b0b 100644
--- a/net/mac80211/iface.c
+++ b/net/mac80211/iface.c
@@ -249,6 +249,8 @@ static int ieee80211_open(struct net_device *dev)
 			local->fif_other_bss++;
 
 		ieee80211_configure_filter(local);
+
+		netif_carrier_on(dev);
 		break;
 	default:
 		res = drv_add_interface(local, &sdata->vif);